### PR TITLE
Integrate concourse

### PR DESCRIPTION
**Pull request by ddadlani**
This is a draft PR to update Voyager with the things we need for Concourse.

All feedback is welcome.

Docs incoming

### CODE REVIEW

File: voyager_test.go [hunk 1]
```diff
@@ -126,16 +130,14 @@ var _ = Describe("Voyager Migration", func() {
 			})
 		})
 		It("SupportedVersion reports the highest supported migration version", func() {
-			migrator = voyager.NewMigrator(logger, db, lockID, source, runner, nil)
-			version, err := migrator.SupportedVersion()
-			Expect(err).ToNot(HaveOccurred())
+			migrator = voyager.NewMigrator(lockID, source, runner, nil)
+			version := migrator.SupportedVersion(logger)
```
Comment (vito) [inline voyager_test.go]:
```
./voyager_test.go:134:12: assignment mismatch: 1 variable but migrator.SupportedVersion returns 2 values
```

File: voyager_test.go [hunk 2]
```diff
@@ -126,16 +130,14 @@ var _ = Describe("Voyager Migration", func() {
...
 			Expect(version).To(Equal(20000))
 		})
 
 		It("Ignores files it can't parse", func() {
-			migrator = voyager.NewMigrator(logger, db, lockID, source, runner, nil)
-			version, err := migrator.SupportedVersion()
-			Expect(err).ToNot(HaveOccurred())
+			migrator = voyager.NewMigrator(lockID, source, runner, nil)
+			version := migrator.SupportedVersion(logger)
```
Comment (vito) [inline voyager_test.go]:
```
./voyager_test.go:140:12: assignment mismatch: 1 variable but migrator.SupportedVersion returns 2 values
```

**Review by vito — CHANGES_REQUESTED** [review]
a couple tests failed to compile, otherwise :+1:

File: voyager_test.go (hunk 1 above)
Comment (ddadlani) [reply voyager_test.go]: 🤦‍♀ I was so focused on functionality, I forgot about tests

**Review by vito — APPROVED** [review]
:+1:

**Comment by ddadlani** [comment]
we still need to change Concourse to use voyager (it doesn't currently use it, the code is embedded in the Concourse codebase)

**Pull request merged**